### ISSUE TITLE
Loss of focus when using MatTextField inside a MatTable

### ISSUE THREAD

**Issue by stefanloerwald** [SamProf/MatBlazor#681]
**Describe the bug**

When defining a callback for `OnFocus` on a `MatTextField` inside a `MatTable`, the text field is not focusable (immediately loses focus after focusing).

**To Reproduce**

1. Create text field in table:
```
<MatTable Items="Items" ShowFooter="false" ShowPaging="false">
    <MatTableHeader>
        <th>Input</th>
    </MatTableHeader>
    <MatTableRow>
        <td>
            <MatTextField TValue="string" OnFocus="@(() => { })" />
        </td>
    </MatTableRow>
</MatTable>

@code {
    private List<int> Items = new List<int> { 0 };
}
```
2. click into the text field to focus it
3. observe the immediate loss of focus

**Blazorfiddle link**
The above code does not show the issue when used on https://blazorfiddle.com/s/ms7z0yai, I suspect that versions are different.

**Expected behavior**
No loss of focus

**Additional context**
Version: 2.7.0-develop-017 and 2.6.2 are both affected

**Comment by erlendp** [comment]
Could be related to the PR (#692) I just created. I've moved the key to the table row, as blazor was destroying the whole row whenever a re-render was triggered.

**Comment by enkodellc** [comment]
Definitely a bug. Resolved with the #692 PR merged into develop branch.

**Comment by Christian-Oleson** [comment]
@enkodellc @stefanloerwald ,
If this bug is resolved with the above PR, can we close this issue?
